src/components/dialogs/parameters/dynamic-security-analysis/dynamic-security-analysis-parameters.tsx

```diff
@@ -0,0 +1,234 @@
+        resolver: yupResolver<DynamicSecurityAnalysisParametersForm>(formSchema),
+    });
+
+    const { reset, handleSubmit, formState } = formMethods;
+
+    const onError = useCallback(
+        (errors: FieldErrors<DynamicSecurityAnalysisParametersForm>) => {
+            if (!errors || isObjectEmpty(errors)) {
+                return;
+            }
+
+            const tabsInError = [];
+            // do not show error when being in the current tab
+            if (errors?.[TAB_VALUES.SCENARIO] && TAB_VALUES.SCENARIO !== tabIndex) {
+                tabsInError.push(TAB_VALUES.SCENARIO);
+            }
+            if (errors?.[TAB_VALUES.CONTINGENCY] && TAB_VALUES.CONTINGENCY !== tabIndex) {
+                tabsInError.push(TAB_VALUES.CONTINGENCY);
+            }
+
+            if (tabsInError.includes(tabIndex)) {
+                // error in current tab => do not change tab systematically but remove current tab in error list
+                setTabIndexesWithError(tabsInError.filter((errorTabIndex) => errorTabIndex !== tabIndex));
+            } else if (tabsInError.length > 0) {
+                // switch to the first tab in the list then remove the tab in the error list
+                setTabIndex(tabsInError[0]);
+                setTabIndexesWithError(tabsInError.filter((errorTabIndex, index, arr) => errorTabIndex !== arr[0]));
+            }
+        },
+        [tabIndex]
+    );
+
+    const onSubmit = useCallback(
+        (newParams: DynamicSecurityAnalysisParametersForm) => {
+            // use updater to set with new parameters
+            updateParameters({
+                [PROVIDER]: newParams.provider,
+                ...newParams.scenario,
+                [CONTINGENCIES_START_TIME]: newParams.contingency.contingenciesStartTime,
+                [CONTINGENCIES_LIST_INFOS]: newParams.contingency.contingencyListInfos,
+            });
+        },
+        [updateParameters]
+    );
+
+    useEffect(() => {
+        if (parameters && provider) {
+            reset({
+                [PROVIDER]: parameters.provider ?? provider,
+                [TAB_VALUES.SCENARIO]: {
+                    [SCENARIO_DURATION]: parameters.scenarioDuration,
+                },
+                [TAB_VALUES.CONTINGENCY]: {
+                    [CONTINGENCIES_START_TIME]: parameters.contingenciesStartTime,
+                    [CONTINGENCIES_LIST_INFOS]: parameters.contingencyListInfos,
+                },
+            });
+        }
+    }, [reset, parameters, provider]);
+
+    const handleTabChange = useCallback((event: SyntheticEvent<Element, Event>, newValue: TAB_VALUES) => {
+        setTabIndex(newValue);
+    }, []);
+
+    useEffect(() => {
+        setHaveDirtyFields(!!Object.keys(formState.dirtyFields).length);
+    }, [formState, setHaveDirtyFields]);
+
+    return (
+        <CustomFormProvider validationSchema={formSchema} {...formMethods}>
+            <Grid sx={{ height: '100%' }}>
+                <Grid
```

Comment:
Culd be mutualized in a compenent and used here and in dynamic-simulation-parameters.tsx

Reply:
DONE! extract into a separate component pour provider, hopefully can be reused in others computation params later